include/drivers/adc.h

```diff
@@ -324,6 +324,54 @@ struct adc_sequence {
 	bool calibrate;
 };
 
+enum adc_threshold_mode {
+	ADC_THR_MODE_NONE,
+	ADC_THR_MODE_RISE,
+	ADC_THR_MODE_FALL,
+	ADC_THR_MODE_MAX,
+};
+
+/**
+ * @brief Structure used to configure threshold interruption.
+ */
+struct adc_threshold_cfg {
+	/**
+	 * ADC channel.
+	 * This is the ADC channel that will be actively used to measure.
+	 */
+	uint8_t channel_id;
+
+	/**
+	 * Threshold identifier.
+	 * This number is used to identify threshold being configured,
+	 * values are ranged:
+	 *   0 .. MAX thresholds supported.
+	 * Single ADC channel can support multiple thresholds, and one threshold
+	 * is associated with a single ADC channel and assertion value.
+	 */
+	uint8_t threshold_id;
+
+	/**
+	 * Threshold value
+	 * Establish the threshold assertion value.
+	 */
+	uint16_t value;
```

Comment:
It's not quite clear to me what the intended units of this is. Digging through the nRF code, It's just listed as volts... Is this to be up the driver implementation?

Reply:
Changing variable name to 'assert_value' and adding mV units, this description is complemented  the one for 'assert_mode'.